lib/node_modules/@stdlib/math/base/special/boxcox1p/test/test.js

```diff
@@ -0,0 +1,164 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var randu = require( '@stdlib/random/base/randu' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var NINF = require( '@stdlib/constants/math/float64-ninf' );
+var EPS = require( '@stdlib/constants/math/float64-eps' );
+var boxcox1p = require( './../lib' );
+
+
+// FIXTURES //
+
+var mediumPositive = require( './fixtures/python/medium_positive.json' );
+var smallPositive = require( './fixtures/python/small_positive.json' );
+var tiny = require( './fixtures/python/tiny.json' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof boxcox1p, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var b;
+
+	b = boxcox1p( NaN, 1.0);
+	t.equal( isnan( b ), true, 'returns NaN' );
+
+	b = boxcox1p( 1.0, NaN );
+	t.equal( isnan( b ), true, 'returns NaN' );
+
+	b = boxcox1p( NaN, NaN );
+	t.equal( isnan( b ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'the function returns `-Infinity` if lambda is negative and `x` is -1.0', function test( t ) {
+	var lambda;
+	var b;
+	var i;
+
+	for ( i = 0; i < 1e3; i++ ) {
+		lambda = -1.0 * ( (randu()*100.0) + 1.0 );
+
+		b = boxcox1p( -1.0, lambda );
+		t.equal( b, NINF, 'returns -Infinity when lambda is negative and `x` is negative one' );
+	}
+	t.end();
+});
+
+tape( 'the function returns `NaN` if `x` is negative', function test( t ) {
+	var b;
+	var x;
+	var y;
+	var i;
+
+	for ( i = 0; i < 1e3; i++ ) {
+		x = -1.0 * ( (randu()*100.0) + 1.0 );
```

Comment:
```suggestion
		x = -1.0 * ( (randu()*100.0) + 1.0 + EPS );
```

`randu()` can return `0`.